### PR TITLE
refactor: Use svg icon in ChannelHeader menu button

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Have menu icon rendered with SVG instead of HTML `<span>` elements. Allow to provide custom icon.

### 🛠 Implementation details

- create `MenuIcon` component used by default by `ChannelHeader`
- add `MenuIcon` prop to `ChannelHeaderProps`

### CODE REVIEW

**Review by petyosi — DISMISSED** [review]
Code looks good, but you will need to document the prop here:

`docs/React/utility-components/channel-header.mdx`

File: src/components/ChannelHeader/ChannelHeader.tsx [hunk 1]
```diff
@@ -23,6 +24,8 @@ export type ChannelHeaderProps = {
   image?: string;
   /** Show a little indicator that the Channel is live right now */
   live?: boolean;
+  /** UI component do display menu icon, defaults to [MenuIcon](https://github.com/GetStream/stream-chat-react/blob/master/src/components/ChannelHeader/ChannelHeader.tsx)*/
```
Comment (ambaldwin) [inline src/components/ChannelHeader/ChannelHeader.tsx:27]: UI component to* - typo here

File: src/components/ChannelHeader/ChannelHeader.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ import type {
   DefaultReactionType,
   DefaultUserType,
 } from '../../types/types';
+import { MenuIcon as DefaultMenuIcon } from './icons';
```
Comment (ambaldwin) [inline src/components/ChannelHeader/ChannelHeader.tsx:18]: This import should go above import { AvatarProps, Avatar as DefaultAvatar } from '../Avatar'; with a new line between it and surrounding imports. We group items by depth and abc.

**Review by ambaldwin — CHANGES_REQUESTED** [review]
This is a great addition. I left a couple comments, and we also need to update our docs with this new prop. docusaurus/docs/React/utility-components/channel-header.mdx.

**Review by ambaldwin — CHANGES_REQUESTED** [review]

**Review by ambaldwin — APPROVED** [review]

**Pull request merged**